### ISSUE TITLE
validateField does not validate nested field

### ISSUE THREAD

**Issue by jansedlon** [jaredpalmer/formik#3252]
### Current Behavior

As stated in the title, `validateField` does not validate nested fields like `user.name`. 

### Expected behavior

It should set the correct error.

### Reproducible example

You can switch between the commented lines. This shows the difference between usage of `validateField` with a not nested field and a nested one.

https://codesandbox.io/s/naughty-frost-013fk?file=/src/App.js

### Suggested solution(s)

It took me soooo many hours when I encountered it till I realized that the bug might actually be in the Formik itself. I found that when the field error is set, it should use `getIn(error, name)` instead of `error[name]`. I'm gonna create a PR with a fix and a test.

```diff
diff --git a/packages/formik/src/Formik.tsx b/packages/formik/src/Formik.tsx
index d952a90..8b2d1b7 100755
--- a/packages/formik/src/Formik.tsx
+++ b/packages/formik/src/Formik.tsx
@@ -513,7 +513,7 @@ export function useFormik<Values extends FormikValues = FormikValues>({
         .then((error: any) => {
           dispatch({
             type: 'SET_FIELD_ERROR',
-            payload: { field: name, value: error[name] },
+            payload: { field: name, value: getIn(error, name) },
           });
           dispatch({ type: 'SET_ISVALIDATING', payload: false });
         });
```

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Formik           |  2.2.6
| React            | 17.0.1
| TypeScript       | 4.2.4
| Browser          | Any
| npm/Yarn         | npm 7.16.0
| Operating System | MacOS 11 Big Sur

**Comment by jansedlon** [comment]
Made a PR. Hopefully everything's correct, this is my first PR for Formik

**Comment by jansedlon** [comment]
I don't want this to close, this one is important

**Comment by jansedlon** [comment]
:(

**Comment by silberistgold** [comment]
I have the same issue at the moment. I am trying to use `validateField` on an indexed field in an array like so: `validateField("friends.2.name")` and it does nothing.
As fas as I can tell your fix would enable this as well, correct @jansedlon ?
Can someone please review, merge and release?

**Comment by jansedlon** [comment]
@silberistgold Havent tried that with arrays but  if `getIn` can deal with arrays then I'm pretty confident that it should work as well

**Comment by mckinleymedia** [comment]
+1

**Comment by Samuranz** [comment]
+1

**Comment by alicansoysal** [comment]
+1

**Comment by Clazherx** [comment]
need this!!! how its going?

**Comment by sirkeng** [comment]
I also just get this problem in `getIn`

**Comment by Clazherx** [comment]
actually the only workaround that i found is validate the entire form instead of each field. is not very UX friendly anyways.

**Comment by lancemcmuffin** [comment]
Note: it appears this bug was fixed as of release 2.2.10, via #3783

**Comment by criveratro** [comment]
> Note: it appears this bug was fixed as of release 2.2.10, via #3783

Can confirm the issue was fixed on 2.2.10!
